app/src/androidTest/java/com/android/streetworkapp/ui/utils/CustomDialogTest.kt

```diff
@@ -0,0 +1,115 @@
+package com.android.streetworkapp.ui.utils
+
+import androidx.compose.material3.Text
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.test.assertHasClickAction
+import androidx.compose.ui.test.assertIsDisplayed
+import androidx.compose.ui.test.assertIsNotDisplayed
+import androidx.compose.ui.test.assertTextEquals
+import androidx.compose.ui.test.junit4.createComposeRule
+import androidx.compose.ui.test.onNodeWithTag
+import androidx.compose.ui.test.performClick
+import org.junit.Rule
+import org.junit.Test
+
+class CustomDialogTest {
+
+  @get:Rule val composeTestRule = createComposeRule()
+
+  private lateinit var showDialog: MutableState<Boolean>
+  private var dialogType = "Type"
+
+  @Test
+  fun isDialogCorrectlyDisplayed() {
+    showDialog = mutableStateOf(false)
+
+    composeTestRule.setContent {
+      CustomDialog(
+          showDialog,
+          dialogType,
+          Content = { Text("Content", modifier = Modifier.testTag("content")) },
+          onSubmit = { /* submit function not called */},
+          onDismiss = { /* dismiss function not called */})
+    }
```

Comment:
Improvement: You can create a function since your tests are almost the same with a few different parameters.
```kotlin
private fun setupCustomDialog(
    initialShowDialog: Boolean,
    onSubmit: () -> Unit = {},
    onDismiss: () -> Unit = {}
) {
//to complete
}
```

Reply:
I implemented something alike :)